asv/benchmarks.py

```diff
@@ -338,8 +344,15 @@ def __init__(self, conf, benchmarks, regex=None):
         self._all_benchmarks = {}
         for benchmark in benchmarks:
             self._all_benchmarks[benchmark['name']] = benchmark
-            if not regex or any(re.search(reg, benchmark['name']) for reg in regex):
-                self[benchmark['name']] = benchmark
+            benchmark['_selected_idx'] = []
+            for idx, param_set in enumerate(
+                    itertools.product(*benchmark['params'])):
+                name = '%s(%s)' % (
+                    benchmark['name'],
+                    ', '.join(param_set))
+                if not regex or any(re.search(reg, name) for reg in regex):
+                    self[benchmark['name']] = benchmark
+                    benchmark['_selected_idx'].append(idx)
```

Comment:
Storing it in `benchmark` I think makes it end up in the json files when saving which is a bit ugly.

Maybe 
`self._benchmark_selection[benchmark['name']] = []`
would be nicer?